[fix] 회원가입 api 변경사항 수정

jusang3057 opened the pull request:
codeVerify함수 위치 변경
UserAuthApi -> UserApi
임시 token삭제
TokenResponse.kt 삭제
student/general register의 header 삭제
verify의 response 삭제

jusang3057 committed:
[Feature] 회원가입 API

개요
* 회원가입 API
작업 내용
* https://www.notion.so/25-03-30-API-1c69ae650b5980cc8324dfc77b59cc01

jusang3057 committed:
[Fix] User API 수정

1. request에 not-null적용
2. 버전구분 : [fun]2 -> [fun]_V2
UserRepository.kt
   verifyCertificationCode: return token:String
CheckResponse.kt
   name : CheckResponse -> Duplicated
   data : BAD_REQUEST -> INVALID_FORMAT
   class : enum class -> sealed class
UserRepositoryImpl.kt
   postGeneralRegister(name, pho...) -> postGeneralRegister(token, nam...)
   postStudentRegister(name, pho...) -> postGeneralRegister(token, nam...)

jusang3057 committed:
[Fix] User API 수정

1. request에 not-null적용
2. 버전구분 : [fun]2 -> [fun]_V2
UserRepository.kt
   verifyCertificationCode: return token:String
CheckResponse.kt
   name : CheckResponse -> Duplicated
   data : BAD_REQUEST -> INVALID_FORMAT
   class : enum class -> sealed class
UserRepositoryImpl.kt
   postGeneralRegister(name, pho...) -> postGeneralRegister(token, nam...)
   postStudentRegister(name, pho...) -> postGeneralRegister(token, nam...)

jusang3057 committed:
[Fix] User API 수정

1. request에 not-null적용

2. 버전구분 : [fun]2 -> [fun]_V2

UserRepository.kt
   verifyCertificationCode: return token:String

CheckResponse.kt
   name : CheckResponse -> Duplicated
   data : BAD_REQUEST -> INVALID_FORMAT
   class : enum class -> sealed class

UserRepositoryImpl.kt
   postGeneralRegister(name, pho...) -> postGeneralRegister(token, nam...)
   postStudentRegister(name, pho...) -> postGeneralRegister(token, nam...)

jusang3057 committed:
[Fix] User API 수정

1. Bearer 추가

jusang3057 committed:
klint관련 수정

jusang3057 committed:
ktlint수정

jusang3057 committed:
ktlint수정

jusang3057 committed:
[fix] api 변경사항 수정

1. codeVerify함수 위치 변경
   UserAuthApi -> UserApi
2. 임시 token삭제
   TokenResponse.kt 삭제
   student/general register의 header 삭제
   verify의 response 삭제

jusang3057 committed:
lint 수정

jusang3057 committed:
Merge branch 'develop' into feature/user-new-signup2

jusang3057 committed:
restore

jusang3057 committed:
Merge remote-tracking branch 'origin/feature/user-new-signup2' into feature/user-new-signup2

jusang3057 committed:
restore

kongwoojin commented on data/src/main/java/in/koreatech/koin/data/api/UserApi.kt:
```diff
@@ -71,13 +71,16 @@ interface UserApi {
     )
+
+    @POST(URLConstant.USER.SMSVERIFY)
+    suspend fun codeVerify(
```
codeVerify를 UserApi로 옮겼으니, UserAuthApi에서는 지워야 할 것 같습니다.

kongwoojin reviewed (changes requested):
고생하셨습니다.
코멘트 확인 해주세요
추가적으로, 인증번호 문자 전송 API 요청 값 확인 부탁드립니다.

jusang3057 committed:
[fix] 2025 04 09 명세서

1. 회원가입 문자 인증번호 전송 제거
2. 회원가입 문자 인증번호 검증 제거
3. 문자/이메일 인증번호 전송 추가
4. 문자/이메일 인증번호 검증 추가
5. 인증 횟수 조회 추가
6. phone_number -> target
7. certification_code -> code

jusang3057 committed:
ktlint수정

kongwoojin commented on data/src/main/java/in/koreatech/koin/data/repository/UserRepositoryImpl.kt:
```diff
@@ -251,25 +259,46 @@ class UserRepositoryImpl @Inject constructor(
         } catch (e: HttpException) {
-            false
+            when (e.code()) {
+                400 -> Verification.INVALID
```
명세서에는 400인 경우만 있는데, 에러 코드를 별도로 분기 처리하신 이유가 있을까요?

kongwoojin commented on data/src/main/java/in/koreatech/koin/data/repository/UserRepositoryImpl.kt:
```diff
@@ -251,25 +259,46 @@ class UserRepositoryImpl @Inject constructor(
-            ""
+            when (e.code()) {
+                400 -> Verification.INVALID
+                409 -> Verification.NOCODE
```
404인것 같습니다

kongwoojin commented on data/src/main/java/in/koreatech/koin/data/repository/UserRepositoryImpl.kt:
```diff
@@ -251,25 +259,46 @@ class UserRepositoryImpl @Inject constructor(
+        }
+    }
+
+    override suspend fun countSMS(target: String): CodeRequestCount {
```
Result를 사용해서
API 정상 호출 시 > API 반환 값 그대로 넘겨주기
API 호출 실패 시 > Result.failure 반환
해도 괜찮을 것 같습니다

kongwoojin commented on domain/src/main/java/in/koreatech/koin/domain/model/user/CodeRequestCount.kt:
```diff
@@ -0,0 +1,8 @@
+package `in`.koreatech.koin.domain.model.user
+
+data class CodeRequestCount(
```
보통 Request가 마지막에 옵니다

kongwoojin replied:
이게 /user/verification/count API에 사용되는거라면 Response가 맞는 것 같습니다.

jusang3057 replied:
> 이게 /user/verification/count API에 사용되는 거라면 응답이 맞는 것 같습니다.
1

kongwoojin commented on data/src/main/java/in/koreatech/koin/data/source/remote/UserRemoteDataSource.kt:
```diff
@@ -97,19 +99,23 @@ class UserRemoteDataSource(
+        userApi.codeVerify(smsVerifyRequest)
+    }
+
+    suspend fun countSMS(target: String): Response<CodeRequestCountResponse> {
```
다른 함수와 다르게 Response로 감싼 이유가 있나요?

kongwoojin commented on data/src/main/java/in/koreatech/koin/data/api/UserApi.kt:
```diff
@@ -66,18 +68,26 @@ interface UserApi {
+    @POST(URLConstant.USER.SMSCOUNT)
+    suspend fun smsCount(
+        @Query("target") target: String
+    ): Response<CodeRequestCountResponse>
```
마찬가지로 Response로 감싼 이유가 있을까요?

kongwoojin reviewed (changes requested):
고생하셨습니다
코멘트 확인해주세요

jusang3057 committed:
코드 수정

1. 단일 에러코드 -> Boolean
2. 오타 수정(409 -> 404)
3. CodeRequsetCount -> Result.success()
4. Response<> 제거

jusang3057 committed:
코드 수정

1. coderequestcount.kt -> codecount.kt

kongwoojin reviewed (approved):
고생하셨습니다!
